src/nni_manager/training_service/reusable/environments/amlEnvironmentService.ts

```diff
@@ -83,8 +83,8 @@ export class AMLEnvironmentService extends EnvironmentService {
     public async refreshEnvironmentsStatus(environments: EnvironmentInformation[]): Promise<void> {
         environments.forEach(async (environment) => {
             const amlClient = (environment as AMLEnvironmentInformation).amlClient;
-                    if (!amlClient) {
-            throw new Error('AML client not initialized!');
+            if (!amlClient) {
+                return Promise.reject('AML client not initialized!');
```

Comment:
BTW, why throw error doesn't work?

Reply:
There is no try catch outside the function, throw this error directly will not be shown in web ui.

Reply:
Did you try it? there are a lot of throw error.

Reply:
yes, I tried it. If we only use throw Error without catch, the error information will only shown in `nnictl log stderr` instead of webui.

Reply:
well, the status is not visible so far. But if we expose it to webui in future, it would be a hole.

Reply:
The `throw new Error` here should work without `try catch` because there is a `catch`  in nnimanager and this promise is called from nnimanager.  It works in other training service. Would you please check why it is not working?